### PR TITLE
Handle IArrayElementReferenceOperation in CFG

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IArrayElementReferenceExpression.cs [hunk 1]
```diff
@@ -949,5 +949,561 @@ public void F(string[] args)
...
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void ArrayElementReference_NoControlFlow()
+        {
+            string source = @"
+class C
+{
+    void M(int[] a1, int[,] a2, int i1, int i2, int i3, int result1, int result2)
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IArrayElementReferenceExpression.cs]: >i2 [](start = 44, length = 2)

Consider changing one of the indices, such as `i2` here, to be a `byte` or some other type that involves a conversion to be used as argument to indexer?

**Comment by jcouv** [comment]
It looks like there is a legitimate test failure (`IOperationTests.ObjectCreationFlow_20`).

**Review by jcouv — APPROVED** [review]
LGTM Thanks

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IArrayElementReferenceExpression.cs (hunk 1 above)
Comment (mavasani) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IArrayElementReferenceExpression.cs]: Sure, I will tweak one of the tests to verify this scenario.

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IObjectCreationExpression.cs [hunk 2]
```diff
@@ -3423,30 +3423,36 @@ public void M(bool b)
...
-        Statements (1)
+        Statements (2)
             IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsInvalid, IsImplicit) (Syntax: 'new C1 { O[ ... 2] = null }')
               Value: 
                 IObjectCreationOperation (Constructor: C1..ctor()) (OperationKind.ObjectCreation, Type: C1, IsInvalid) (Syntax: 'new C1 { O[ ... 2] = null }')
                   Arguments(0)
                   Initializer: 
                     null
 
+            IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsInvalid, IsImplicit) (Syntax: 'O')
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IObjectCreationExpression.cs]: >IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsInvalid, IsImplicit) (Syntax: 'O') [](start = 12, length = 101)

This change in evaluation order feels unexpected. Could you please sync-up with @333fred and confirm order of evaluation in IL.

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 3)

**Review by 333fred — APPROVED** [review]
:shipit:

**Pull request merged**